server/game/basecard.js

```diff
@@ -467,6 +471,10 @@ class BaseCard {
         this.game.raiseEvent('onCardTraitChanged', { card: this });
     }
 
+    getTraits() {
+        return _.keys(_.omit(this.traits, trait => trait < 1));
```

Comment:
Not sure what this is trying to do, but I suspect it isn't actually doing what you want it to

Reply:
It removes any trait from traits that has value 0, then takes the keys from the trait object and puts it in an array. It returns ```[ 'king', 'lord' ]``` for Robert Baratheon for example. The omit is necessary for traits that have been gained (creates new key/value pair in traits) and then lost (value is set to 0, but the key/value pair remains), for example through a crown attachment.

Reply:
Ah yes, traits are a refcount aren't they... Should probably get fixed to be removed if it hits 0 but we can deal with that separately. I don't like how this looks but it'll probably have to do for now